scanner/src/test/kotlin/scanners/scancode/ScanCodeResultParserTest.kt

```diff
@@ -778,4 +778,19 @@ class ScanCodeResultParserTest : WordSpec({
             mapUnknownIssues(summary.issues.toMutableList()) shouldBe false
         }
     }
+
+    "replaceLicenseKeys()" should {
+        "Properly handle redundant replacements" {
+            val expression = "public-domain"
+            val replacements = listOf(
+                LicenseKeyReplacement("public-domain", "LicenseRef-scancode-public-domain"),
+                LicenseKeyReplacement("public-domain", "LicenseRef-scancode-public-domain")
+            )
+
+            val result = replaceLicenseKeys(expression, replacements)
+
+            // TODO: This illustrates a bug which should be fixed.
+            result shouldBe "LicenseRef-scancode-LicenseRef-scancode-public-domain"
```

Comment:
Just a side note: I really wish Kotest would support this kind of TDD style and allow to write the really expected value here without breaking the test.

Reply:
You could just ignore the test to achieve that. But what I like about the approach taken here is that the diff of the commit that fixes the bug nicely shows the effect of the bug fix.

Reply:
I just tried `shouldFail` and it looks like `shouldFail { result shouldBe "LicenseRef-scancode-public-domain" }`.
Actually I do like `result shouldBe "LicenseRef-scancode-LicenseRef-scancode-public-domain"` more, because it give more information to the reader, e.g. it shows exactly what the buggy behavior produces. @sschuberth If you're ok I'd leave it as is.